src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,187 @@
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = newNode;
+        } else {
+            getLastNode(bucketIndex).next = newNode;
+        }
+    }
+
+    private Node<K, V> getLastNode(int bucketIndex) {
+        Node<K, V> current = table[bucketIndex];
+        while (current.next != null) {
+            current = current.next;
+        }
+        return current;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public V remove(K key) {
+        Node<K, V> neededNode = getNode(key);
+        if (neededNode == null) {
+            return null;
+        }
+        int bucketIndex = getBucketIndex(neededNode.hash);
+        Node<K, V> previous = null;
+        Node<K, V> current = table[bucketIndex];
+        while (current != null) {
+            if (neededNode.equals(current)) {
+                V oldValue = current.value;
+                if (previous != null) {
+                    previous.next = current.next;
+                } else {
+                    table[bucketIndex] = current.next;
+                }
+                size--;
+                return oldValue;
+            }
+            previous = current;
+            current = current.next;
+        }
+        return null;
+    }
+
+    @Override
+    public boolean containsKey(K key) {
+        return getNode(key) != null;
+    }
+
+    @Override
+    public boolean containsValue(V value) {
+        for (Node<K, V> node : table) {
+            Node<K, V> current = node;
+            while (current != null) {
+                if (current.value.equals(value)) {
+                    return true;
+                }
+                current = current.next;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return getSize() == 0;
+    }
+
+    private int calculateHash(K key) {
+        return key != null ? Objects.hash(key) : 0;
+    }
+
+    private int getBucketIndex(int hash) {
+        return Math.abs(hash % capacity);
+    }
+
+    private void transferIfNeed() {
+        if (size == threshold) {
+            capacity *= MULTIPLIER_FOR_INCREASE_CAPACITY;
+            threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+            Node<K, V>[] oldTable = table;
+            table = new Node[capacity];
+            size = 0;
+            replaceNodes(oldTable);
+        }
+    }
+
+    private void replaceNodes(Node<K, V>[] oldTable) {
+        for (Node<K, V> node : oldTable) {
+            Node<K, V> current = node;
+            while (current != null) {
+                put(current.key, current.value);
+                current = current.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
```

Comment:
Private methods and classes should be after public ones in your class.

Reply:
Remove hash from constructor